src/proc/syncer-processing-block.h

```diff
@@ -2,28 +2,31 @@
 // Copyright(c) 2015 Intel Corporation. All Rights Reserved.
 
 #pragma once
-#include "types.h"
-#include "archive.h"
 
 #include <stdint.h>
 #include <vector>
 #include <mutex>
 #include <memory>
 
+#include "types.h"
+#include "archive.h"
+#include "option.h"
+
 namespace librealsense
 {
     class processing_block;
     class timestamp_composite_matcher;
     class syncer_process_unit : public processing_block
     {
     public:
-        syncer_process_unit();
+        syncer_process_unit(bool_option* is_enabled_opt = nullptr);
```

Comment:
Externally this is wrapped with shared_ptr, so if somehow the external instance is being reset then it is a potentially dangling prt. For instance deregister_option